docs/TextRegion.md

```diff
@@ -0,0 +1,86 @@
+	* Text will be rendered automatically after closing the context.
+	* When used as a context manager, you can change all text styling parameters within that context, and they will be used by the added text, but won't leak to the surroundings
+* Alternatively, eg. for filling a single column of text with the already existing settings, just use the region instance as is. In that case, you'll have to explicitly use the `render()` method after adding the text.
+* Within a region, paragraphs can be inserted. The primary purpose of a paragraph is to apply a different horizontal alignment than the surrounding text. It is also possible to apply margins to the top and bottom of each paragraph.
+
+![](tcols-paragraphs.png)
+
+The graphic shows shows the relationship of page, text areas and paragraphs (with varying alignment) for the example of a two-column layout.
+
+
+### Text Start Position ###
+
+When rendering, the vertical start position of the text will be at the lowest one out of the current y position, the top of the region (if it has a defined top), or the top margin of the page. The horizontal start position will be either at the current x position if that lies within the boundaries of the region/column, or at the left edge of the region. In both horizontal and vertical positioning, regions with multiple columns may follow additional rules and restrictions.
+
+
+### Interaction between Regions ###
+
+Several region instances can exist at the same time. But only one of them can act as context manager at any given time. It is not currently possible to activate them recursively. But it is possible to use them intermittingly. This will probably most often make sense between a columnar region and a table or a graphic. You may have some running text ending at a given height, then insert a table/graphic, and finally continue the running text at the new height below the table within the existing column(s).
+
+
+### Common parameters ###
+
+All types of text regions have the following constructor paramters in common:
+
+* text (str, optional) - text content to add to the region. This is a convenience parameter for cases when all text is available in one piece, and no partition into paragraphs (possibly with different parameters) is required. (Default: None)
+* align (Align/str, optional) - the horizontal alignment of the text in the region. (Default: Align.L)
+* line_height (float, optional) - This is a factor by which the line spacing will be different from the font height. It works similar to the attribute of the same name in HTML/CSS. (default: 1.0)
+* print_sh (bool, optional) - Treat a soft-hyphen (\\u00ad) as a printable character, instead of a line breaking opportunity. (Default: False)
+* skip_leading_spaces (default: False) - This flag is primarily used by `write_html()`, but may also have other uses. It removes all space characters at the beginning of each line.
+* wrapmode (default "WORD") - 
```

Comment:
This really look like the docstrings for the constructor of `ParagraphCollectorMixin` 🙂
Maybe it should be transformed into its actual docstring,
and a link could be made there to <https://py-pdf.github.io/fpdf2/fpdf/text_region.html#fpdf.text_region.ParagraphCollectorMixin>?

The main benefit I see of doing that, is that if changes are made to `ParagraphCollectorMixin` parameters in the future, it is much more likely that developpers think about updating the docstring that to update this documentation section.

But if you think that it makes the documentation really clearer to have this list of settings described there,
I'm OK to keep it that way. Then we could maybe add a link in `ParagraphCollectorMixin` docstring pointing to this documentation section.

Reply:
Just that `ParagraphCollectorMixin` isn't part of the public API, so I'm quite reluctant to explicitly link to its docstring. I also don't know yet where those parameters will actually end up once the system is more or less complete. We may revisit this question later.

Reply:
Alright!